### ISSUE TITLE
CLI requires an unsupported version of Node and the requirement is not documented

### ISSUE THREAD

**Issue by caraya** [foundation/foundation-cli#140]
Node 10 went end-of-life in April and it shouldn't be the solution to work around the natives errors when trying to run the CLI in Node 12 or 14 (tried with 12.22.1 and 14.16..1). 

The author of natives warned over a year ago that the module would break. Doesn't help that the last version of the fork in use was released over a year ago

The need to use a specific version of Node is not documented in the CLI docs and can cause additional, and unnecessary confusion.

**Comment by justingit** [comment]
Installing Node 10 doesn't work using Homebrew anymore either, 

`
justin@m1 testffe % brew install node@10                    
Warning: node@10 has been deprecated because it is not supported upstream!
Error: node@10: no bottle available!
You can try to install from source with:
  brew install --build-from-source node@10
Please note building from source is unsupported. You will encounter build
failures with some formulae. If you experience any issues please create pull
requests instead of asking for help on Homebrew's GitHub, Twitter or any other
official channels.
justin@m1 testffe % 
`
Installing from source also fails with errors.

**Comment by justingit** [comment]
(or via nvm on a Macbook M1).

**Comment by justingit** [comment]
@caraya, just trying to help with your issue - I don't really think that the underlying issue you bring up is ever going to be fixed, unfortunately. As you noted, they've had a long time to address it, but have not.

**Comment by caraya** [comment]
@justingit Thank you but my issue has nothing to do with Apple Silicon and everything with them removing or working around a dependency that they were warned would break.  It may be solved or not but I'm pretty sure I'm not the only one who worries about being forced to use unsupported versions of Node.

I guess I'll have to wait and see if they choose to do anything about it

**Comment by nanawel** [comment]
Just to be sure, to sum up: there's currently _no way_ to use `foundation-cli` anymore, right?

Installing it on **Node 10** breaks with `error ansi-regex@6.0.0: The engine "node" is incompatible with this module. Expected version ">=12". Got "11.15.0"`.

Installing it on **Node 12+** succeeds, but then using it fails with:

```
fs.js:36
} = primordials;
    ^

ReferenceError: primordials is not defined
```

**Comment by benthien** [comment]
I installed node 11.15.0 on my PC running Windows 10. I then installed foundation-cli with no problems. Both foundation watch and foundation build are working.

**Comment by caraya** [comment]
@benthien The problem is that anything below Node 12 is unsupported and will not get any updates or patches, even security updates. The fact that you can do it doesn't make it any more secure and any less prone to conflict from other packages that may have version requirements as @nanawel pointed out in his comment. 

> Installing it on Node 10 breaks with error ansi-regex@6.0.0: The engine "node" is incompatible with this module. Expected version ">=12". Got "11.15.0".

So while foundation may work on Node 11 for your particular case, it is not guaranteed to work everywhere.

It is also important to note that Node 11 was a short-lived development version and was never meant to be actively used in production work. Node 11 is also EOL as far as I know

**Comment by fburleigh** [comment]
I was able to install foundation-cli with Homebrew'd node 17.1 on my Mac with M1 (Silicon) chip. This thread told me that there would be problems. I gave it a try anyway. The thread is right. ;-)

Analysis of the "new" command told me the jobs of "new" are: check if there's a GIT it can use, gather information about the project a user wants to create, make a directory for that project, use GIT to clone one of the Foundation for Sites template repos (then delete the .git directory) and, finally, use NPM's API to process the package.json file to install dependencies.

Some details:

1. NPM from version 8 lacks the "programmatic API" that the Foundation CLI needs to perform NPM's work. So you're left with a package_lock.json file in the parent of the project directory, and no installed dependencies in the project directory. NPM install will work sort of okay nonetheless.
2. But because version requirements require older packages, there are zillions of deprecation (and security) warnings and in the end a failed installation (probably node-sass).
3. And the project made by the CLI would like to use the deprecated node-sass (and hence LibSass) processor instead of the newer Dart processor.

Since I've seen Dart work on the Foundation code base, I suspect one could retrofit that info gulpfile.js and then make some progress. But I can't be sure of that because none of the Foundation projects use every Foundation block.

So, that's one person's experience.
